core/src/blocktree.rs

```diff
@@ -2188,6 +2233,97 @@ pub mod tests {
         Blocktree::destroy(&ledger_path).expect("Expected successful database destruction");
     }
 
+    #[test]
+    pub fn test_completed_blobs_signal() {
+        // Initialize ledger
+        let ledger_path = get_tmp_ledger_path("test_completed_blobs_signal");
+        let (ledger, _, recvr) = Blocktree::open_with_signal(&ledger_path).unwrap();
+        let ledger = Arc::new(ledger);
+
+        let entries_per_slot = 10;
+        // Create blobs for slot 0
+        let (blobs, _) = make_slot_entries(0, 0, entries_per_slot);
+
+        // Insert all but the first blob in the slot, should not be considered complete
+        ledger
+            .insert_data_blobs(&blobs[1..entries_per_slot as usize])
+            .unwrap();
+        assert!(recvr.try_recv().is_err());
+
+        //Insert first blob, slot should now be considered complete
+        ledger.insert_data_blobs(once(&blobs[0])).unwrap();
+        assert_eq!(recvr.try_recv().unwrap(), vec![0]);
+    }
+
+    #[test]
+    pub fn test_completed_blobs_signal_orphans() {
```

Comment:
I'm having a hard time grokking these tests. I don't understand the magic numbers.  Usually I make a minimal change to test an edge case or use the number `42` to mean "arbitrary". Here I see 10, 5, 2 and while 10=5*2, I don't think that's relevant.  Is there a way the first test can be written as a specialization of this one?   Or perhaps a helper function so that we can see just the one number change instead of the big copy-paste?

Reply:
I added a helper function to generate a chain of slots, to better communicate that the numbers are arbitrary numbers, and that the relationship to note is that the slots chain to each other.